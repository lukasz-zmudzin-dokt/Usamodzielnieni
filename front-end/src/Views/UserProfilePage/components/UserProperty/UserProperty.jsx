import React from "react";
import { Row, Col, ListGroupItem } from "react-bootstrap";

const UserProperty = props => {
  const { names, property, user } = props;
  return (
    <ListGroupItem className="property">
      <Row>
        <Col xs="12" sm="auto">
          <div className="property__title">{names.property[property]}:</div>
        </Col>
        <Col>{user[property]}</Col>
      </Row>
    </ListGroupItem>
  );
};
export default UserProperty;
