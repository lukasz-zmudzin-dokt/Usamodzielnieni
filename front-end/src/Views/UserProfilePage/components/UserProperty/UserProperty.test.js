import React from "react";
import { render } from "@testing-library/react";
import UserProperty from "Views/UserProfilePage/components/UserProperty";

const user = {
  username: "user1",
  role: "common",
  firstName: "Jan",
  lastName: "Kowalski",
  email: "jan.kowalski@pw.edu.pl",
  phoneNumber: "+48123456789",
};

const names = {
  role: {
    admin: "Administrator",
    employer: "Pracodawca",
    common: "Podopieczny",
  },
  property: {
    username: "Nazwa użytkownika",
    role: "Rola",
    firstName: "Imię",
    lastName: "Nazwisko",
    email: "E-mail",
    phoneNumber: "Numer telefonu",
  },
};

describe("UserProperty ", () => {
  it("should render correctly", () => {
    const { container } = render(<UserProperty user={user} names={names} />);
    expect(container).toMatchSnapshot();
  });
});
