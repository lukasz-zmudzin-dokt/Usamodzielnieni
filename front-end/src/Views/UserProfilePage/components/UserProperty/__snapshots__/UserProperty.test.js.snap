// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProperty  should render correctly 1`] = `
<div>
  <div
    class="property list-group-item"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-auto col-12"
      >
        <div
          class="property__title"
        >
          :
        </div>
      </div>
      <div
        class="col"
      />
    </div>
  </div>
</div>
`;
